pkg/agent/controller/agent.go

```diff
@@ -228,94 +304,167 @@ func (a *Controller) serviceImportLister(transform func(si *mcsv1a1.ServiceImpor
 	return retList
 }
 
-func (a *Controller) serviceExportToServiceImport(obj runtime.Object, numRequeues int, op syncer.Operation) (runtime.Object, bool) {
-	svcExport := obj.(*mcsv1a1.ServiceExport)
+func (a *Controller) serviceExportUploadTransform(serviceExportObj runtime.Object, _ int, op syncer.Operation) (runtime.Object, bool) {
+	localServiceExport := serviceExportObj.(*mcsv1a1.ServiceExport)
 
-	klog.V(log.DEBUG).Infof("ServiceExport %s/%s %sd", svcExport.Namespace, svcExport.Name, op)
+	klog.V(log.DEBUG).Infof("Local ServiceExport %s/%s %sd", localServiceExport.Namespace, localServiceExport.Name, op)
 
+	brokerServiceExport := a.newServiceExport(localServiceExport.Name, localServiceExport.Namespace)
```

Comment:
Consider moving object creation closer to where it's used?

Reply:
we do need to early break and return it on deletion